themes/src/main/resources/theme/base/login/login-otp.ftl

```diff
@@ -3,7 +3,7 @@
     <#if section="header">
         ${msg("doLogIn")}
     <#elseif section="form">
-        <form id="kc-otp-login-form" class="${properties.kcFormClass!}" action="${url.loginAction}"
+        <form id="kc-otp-login-form" class="${properties.kcFormClass!}" onsubmit="login.disabled = true; return true;" action="${url.loginAction}"
```

Comment:
How exactly does this prevent additional submission? What is `login` in this case? Where is it defined?

Reply:
Imo it's referencing the `loginButton` by its name attribute.

```
<#macro loginButton>
  <@buttons.actionGroup>
    <@buttons.button id="kc-login" name="login" label="doLogIn" class=["kcButtonPrimaryClass", "kcButtonBlockClass"] />
  </@buttons.actionGroup>
</#macro>
```

I did not question this, but I see your point. 
This is an implicit / legacy way of inline event handlers in JavaScript, which tends to be brittle and probably even needs an `unsave-eval` setting in CSP to work. 

https://stackoverflow.com/questions/57048009/why-is-it-possible-to-access-form-values-in-the-inline-event-handler

1) I can pull the logic into a  `<script>` tag or JavaScript file if you agree. 
2) I think I should do that for every occasion using the login button.

WDYT

Reply:
Wow, I have never in my tenure seen this functionality, TIL. As far as I can tell this is not even documented on MDN.

I agree with the proposed enhancements, but we'll likely handle this as part of our effort to clean up the CSP (see #16277), so I don't consider it a reason to hold up this enhancement.